### PR TITLE
[Indexer][testing][2/n] Create framework for test cases.

### PR DESCRIPTION
**Pull request by larry-aptos**
## Description

* Framework change to support test cases with
  * Step::Setup
  * Step::Action

* Next step
  * Impl the txn generation from test case

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [x] Other (specify)

## How Has This Been Tested?

ut

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by larry-aptos** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/13853?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13853** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/13853?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#13842** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/13842?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @larry-aptos and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ecosystem/indexer-grpc/indexer-transaction-generator/src/test_case.rs [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+/// It's to facilitate the validation of the test case.
+/// Error vs UNKNOWN:
+///  - If a PathBuf can be regex matched to a setup or action step, but cannot be parsed as a folder, it's an error.
+///  - If a PathBuf cannot be regex matched to a setup or action folder, it's UNKNOWN.
+#[derive(PartialEq, Eq, Debug, Hash)]
+pub enum Step {
+    Setup((PathBuf, u64)),
+    Action((PathBuf, u64)),
+    /// This is to handle the case when the step is not a setup or action folder.
+    UNKNOWN,
```
Comment (banool) [inline ecosystem/indexer-grpc/indexer-transaction-generator/src/test_case.rs:98]: Out of curiosity when do we expect additional unknown folders like this? Might be good to explain in the comments.

File: ecosystem/indexer-grpc/indexer-transaction-generator/src/test_case.rs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+impl Step {
+    fn priority(&self) -> u64 {
+        match self {
+            Step::Setup((_, index)) => *index,
+            Step::Action((_, index)) => *index + u64::MAX / 2,
```
Comment (banool) [inline ecosystem/indexer-grpc/indexer-transaction-generator/src/test_case.rs:105]: What if there are 9223372036854775808 setup steps? heheh /troll

File: ecosystem/indexer-grpc/indexer-transaction-generator/src/tests/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Testing Files for the framework.
+
+__Do not put actual test case here__
```
Comment (banool) [inline ecosystem/indexer-grpc/indexer-transaction-generator/src/tests/README.md:3]: Maybe mention what they should do instead.

**Review by banool — APPROVED** [review]
Nice looking great, much cleaner than before!

File: ecosystem/indexer-grpc/indexer-transaction-generator/src/tests/README.md (hunk 3 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-transaction-generator/src/tests/README.md]: done.

File: ecosystem/indexer-grpc/indexer-transaction-generator/src/test_case.rs (hunk 1 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-transaction-generator/src/test_case.rs:100]: done done done.

like `README.md` or `.ds_store` to ignore.

**Pull request merged**